datadog_checks_dev/datadog_checks/dev/tooling/manifest_validator/common/validator.py

```diff
@@ -177,36 +178,86 @@ def validate(self, check_name, decoded, _):
+    """
+
+    MANIFEST_VERSION_PATH = "manifest_version"
 
-    FIELDS_NOT_ALLOWED_TO_CHANGE = {
+    IMMUTABLE_FIELD_PATHS = {
         V1: ("integration_id", "display_name", "guid"),
-        V2: ("id", "source_type_name", "app_id"),
+        V2: (
+            "app_id",
+            "assets/integration/id",
+            "assets/integration/source_type_name",
+        ),
+    }
+
+    SHORT_NAME_PATHS = {
+        V1: (
+            "assets/dashboards",
+            "assets/monitors",
+            "assets/saved_views",
+        ),
+        V2: (
+            "assets/dashboards",
+            "assets/monitors",
+            "assets/saved_views",
+        ),
     }
 
     def validate(self, check_name, decoded, fix):
-        manifest_fields_changed = content_changed(file_glob=f"{check_name}/manifest.json")
+        # Check if previous version of manifest exists
+        # If not, this is a new file so this validation is skipped
+        try:
+            previous = git_show_file(path=f"{check_name}/manifest.json", ref="master")
+            previous_manifest = JSONDict(json.loads(previous))
+        except Exception:
+            self.result.messages['info'].append(
+                "  skipping check for changed fields: integration not on default branch"
+            )
+            return
 
         # Skip this validation if the manifest was updated from 1.0.0 -> 2.0.0
-        manifest_upgrade_regex = re.compile(r'\+\s+"manifest_version": "2.0.0",')
-        version_upgraded = manifest_upgrade_regex.findall(manifest_fields_changed)
+        current_manifest = decoded
+        if (
+            previous_manifest[self.MANIFEST_VERSION_PATH] == "1.0.0"
+            and current_manifest[self.MANIFEST_VERSION_PATH] == "2.0.0"
+        ):
+            self.result.messages['info'].append("  skipping check for changed fields: manifest version was upgraded")
+            return
 
-        if 'new file' not in manifest_fields_changed and not version_upgraded:
-            for field in self.FIELDS_NOT_ALLOWED_TO_CHANGE[self.version]:
-                if field in manifest_fields_changed:
-                    output = f'Attribute `{field}` is not allowed to be modified. Please revert to original value'
+        # Check for differences in immutable attributes
+        immutable_attribute_changed = False
+        for key_path in self.IMMUTABLE_FIELD_PATHS[self.version]:
+            previous_value = previous_manifest.get_path(key_path)
+            current_value = current_manifest.get_path(key_path)
+
+            if previous_value != current_value:
+                output = f'Attribute `{current_value}` at `{key_path}` is not allowed to be modified. Please revert it \
+to the original value `{previous_value}`.'
+                immutable_attribute_changed = True
+                self.fail(output)
+
+        # If already failed, stop validations with failure messages gathered above
+        if immutable_attribute_changed:
+            return
```

Comment:
Should we move this down below the check for sshort_names? If I understand right, as-is, we'd fail if something like `integration_id` was modified but we wouldn't show the errors regarding short_name

Reply:
I removed this completely so the test can run until the end.